### PR TITLE
Delete and settings module

### PR DESCRIPTION
**Pull request by a2batic**

### CODE REVIEW

**Comment by gnehapk** [comment]
@a2batic Please add  github issue to it.

File: src/commons/factories/user-factory.js [hunk 1]
```diff
@@ -98,6 +98,29 @@
                 utils.checkErrorCode(e);
             });
         };
+
+        vm.doActionOnUser = function(username) {
+            var url, actionRequest, request;
+
+
+            url = config.baseUrl + "users/" + username;
+
```
Comment (gnehapk) [inline src/commons/factories/user-factory.js:107]: @a2batic Remove unnecessary empty lines.

File: src/commons/factories/user-factory.js [hunk 2]
```diff
@@ -98,6 +98,29 @@
                 utils.checkErrorCode(e);
             });
         };
+
+        vm.doActionOnUser = function(username) {
```
Comment (gnehapk) [inline src/commons/factories/user-factory.js:102]: @a2batic As we have discussed, this should be renamed to deleteUser name.

File: src/commons/factories/user-factory.js [hunk 3]
```diff
@@ -98,6 +98,29 @@
...
+            var url, actionRequest, request;
+
+
+            url = config.baseUrl + "users/" + username;
+
+
+            actionRequest = {
```
Comment (gnehapk) [inline src/commons/factories/user-factory.js:109]: @a2batic Rename it to deleteUserRequest.

File: src/commons/services/auth-manager.js [hunk 4]
```diff
@@ -31,7 +31,7 @@
             isAuthenticated: isAuthenticated,
             setUserRole: setUserRole,
             getUserRole: getUserRole,
-            clearUserRole: clearUserRole
+            clearUserRole: clearUserRole,
```
Comment (gnehapk) [inline src/commons/services/auth-manager.js:34]: @a2batic Remove comma from the end.

File: src/commons/stores/user-store.js [hunk 5]
```diff
@@ -129,9 +119,23 @@
             return deferred.promise;
         };
 
+        store.doActionOnUser = function(username) {
```
Comment (gnehapk) [inline src/commons/stores/user-store.js:122]: @a2batic Rename it to deleteUser.

File: src/modules/users/delete-user/delete-user.js [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        }, {
+            "name": "Yes",
+            "classname": "btn-primary",
+            "onCall": vm.confirmModal
+        }];
+
+        /**
+         * @name cancelModal
+         * @desc cancels the modal
+         * @memberOf StopVolumeController
```
Comment (gnehapk) [inline src/modules/users/delete-user/delete-user.js:37]: @a2batic Replace StopVolumeController with deleteUserController.

File: src/modules/users/edit-user/edit-user.html [hunk 7]
```diff
@@ -100,8 +94,8 @@
                     </div>
                     <div class="radio radio-class">
                         <label>
-                            <input type="radio" name="role" id="role3" value="limited" ng-model="editUserCntrl.user.role"> <strong>Limited User</strong>
-                            <br> The Limited user role permits the user to access Tendrl in read-only mode, which means the user cannot actively manage storage. This role is ideal for users interested in monitoring the storage.
+                            <input type="radio" name="role" id="role3" value="limited" ng-model="editUserCntrl.user.role"> <strong>Read-only User</strong>
+                            <br> The Read-only user role permits the user to access Tendrl in read-only mode, which means the user cannot actively manage storage. This role is ideal for users interested in monitoring the storage.
```
Comment (gnehapk) [inline src/modules/users/edit-user/edit-user.html:98]: @a2batic As user is not allowed to update role. Please remove the html for it.
.

File: src/modules/users/user-list/users.js [hunk 8]
```diff
@@ -51,6 +59,37 @@
...
+                backdrop: "static",
+                templateUrl: "/modules/users/delete-user/delete-user.html",
+                controller: "deleteUserController",
+                controllerAs: "vm",
+                size: "md",
+                resolve: {
+                    selectedUser: function() {
+                        return {
+                            username: username,
+                            userList: vm.userList
```
Comment (gnehapk) [inline src/modules/users/user-list/users.js:79]: @a2batic I don't think, you need to pass userList now.

File: src/modules/users/delete-user/delete-user.js [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+        function confirmModal() {
+            userStore.doActionOnUser(selectedUser.username)
+                .then(function(data) {
+                    vm.closeModal();
+                    userStore.getUserList()
+                        .then(function(data) {
+                            if (data !== null) {
+                                selectedUser.userList = data;
+                            }
+                            $rootScope.$broadcast("UpdatedUserList", selectedUser.userList);
```
Comment (a2batic) [inline src/modules/users/delete-user/delete-user.js:68]: @gnehapk, we would require userlist here, it is been watched here

Comment (gnehapk) [reply src/modules/users/delete-user/delete-user.js:68]: @a2batic yes. But we don't need modal to pass user list. We can directly broadcast data here.

**Pull request closed without merging**